web/src/compose_ui.js

```diff
@@ -354,12 +355,15 @@ export function format_text($textarea, type, inserted_content) {
     let text = $textarea.val();
     const selected_text = range.text;
 
-    // Remove new line and space around selected text.
-    const left_trim_length = range.text.length - range.text.trimStart().length;
-    const right_trim_length = range.text.length - range.text.trimEnd().length;
+    // Remove new line and space around selected text, except for list formatting.
```

Comment:
This could really benefit from the comment explaining why this is turned off for list formatting; the fact that it's disabled for those is clear from the code.

Reply:
It's so that we also consider any newline selected at the start / end, to format that line too. In this example, the `|` denotes the start and end of selection, and on formatting this selection, all 3 lines get formatted:
```
selected from end of this line|
through this line
|to the start of this line
```

Have explained this in the comment